src/order/filter/n_ary.lean

```diff
@@ -49,6 +50,29 @@ def map₂ (m : α → β → γ) (f : filter α) (g : filter β) : filter γ :=
 lemma image2_mem_map₂ (hs : s ∈ f) (ht : t ∈ g) : image2 m s t ∈ map₂ m f g :=
 ⟨_, _, hs, ht, subset.rfl⟩
 
+lemma map_prod_eq_map₂ {f : filter α} {g : filter β} (m : α → β → γ) :
+  filter.map (λ p : α × β, m p.1 p.2) (f ×ᶠ g) = map₂ m f g :=
+begin
+  rw filter.ext_iff,
+  intro s,
+  split,
```

Comment:
You should add some `{}` after the split to separate the proofs for each direction